packages/ui/src/components/Checkbox/index.tsx

```diff
@@ -69,7 +69,10 @@ const CheckboxInput = styled('input', {
     cursor: not-allowed;
   }
 
-  &:checked + ${StyledIcon}, &[aria-checked='mixed'] + ${StyledIcon} {
+  &:checked:not(:disabled)
+    + ${StyledIcon},
+    &[aria-checked='mixed']:not(:disabled)
+    + ${StyledIcon} {
```

Comment:
Maybe we can simplify it:

```suggestion
&:not(:disabled) {  
  &:checked + ${StyledIcon}, [aria-checked='mixed'] ${StyledIcon} {
    ...
  }
}
```